test/models/bot/alegre_test.rb

```diff
@@ -114,16 +131,74 @@ def setup
     end
   end
 
-  test "should add relationships" do
+  def self.extract_project_medias_from_context(search_result)
+    # We currently have two cases of context:
+    # - a straight hash with project_media_id
+    # - an array of hashes, each with project_media_id
+    context = search_result.dig('_source', 'context')
+    pms = []
+    if context.kind_of?(Array)
+      context.each{ |c| pms.push(c.with_indifferent_access.dig('project_media_id')) }
+    elsif context.kind_of?(Hash)
+      pms.push(context.with_indifferent_access.dig('project_media_id'))
+    end
+    Hash[pms.flatten.collect{|pm| [pm.to_i, search_result.with_indifferent_access.dig('_score')]}]
+  end
+
+  test "should extract project medias from context" do
+    assert_equal Bot::Alegre.extract_project_medias_from_context({"_score" => 2, "_source" => {"context" => {"project_media_id" => 1}}}), {1=>2}
+  end
+
+  test "should relate project media to similar items" do
     p = create_project
-    pm1 = create_project_media project: p
-    pm2 = create_project_media project: p
-    pm3 = create_project_media project: p
-    create_relationship source_id: pm3.id, target_id: pm2.id
-    Bot::Alegre.add_relationships(pm1, [pm2.id])
+    pm1 = create_project_media project: p, is_image: true
+    pm2 = create_project_media project: p, is_image: true
+    pm3 = create_project_media project: p, is_image: true
+    create_relationship source_id: pm2.id, target_id: pm1.id
+    Bot::Alegre.stubs(:request_api).returns({
+      "result" => [
+        {
+          "id" => 1,
+          "sha256" => "1782b1d1993fcd9f6fd8155adc6009a9693a8da7bb96d20270c4bc8a30c97570",
+          "phash" => 17399941807326929,
+          "url" => "https:\/\/www.gstatic.com\/webp\/gallery3\/1.png",
+          "context" => [{
+            "team_id" => pm2.team.id.to_s,
+            "project_media_id" => pm2.id.to_s
+          }],
+          "score" => 1.0
+        }
+      ]
+    })
+    Bot::Alegre.stubs(:media_file_url).with(pm3).returns("some/path")
```

Comment:
Please make sure to `unstub` at the end of the test (or at once in `def teardown`) because it doesn't happen automatically in this version of the library. Something like: `Bot::Alegre.unstub(:request_api)`.